diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08e9a79..c1e2fed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,8 +51,6 @@ find_package(PkgConfig)
 pkg_check_modules(DEPS REQUIRED ${NEEDED_PKG} gthread-2.0 gtksourceview-3.0>=3.10 libqalculate)
 add_definitions(${DEPS_CFLAGS})
 
-add_subdirectory(libqalculatenasc)
-
 link_directories(${DEPS_LIBRARY_DIRS})
 link_directories(${CMAKE_BINARY_DIR})
 
@@ -95,13 +93,10 @@ OPTIONS
 include_directories(${CMAKE_BINARY_DIR})
 include_directories(${CMAKE_SOURCE_DIR})
 include_directories(${CMAKE_SOURCE_DIR}/libqalculatenasc)
-add_executable(nasc ${VALA_C})
-add_dependencies(nasc
-    qalculatenasc)
+add_executable(nasc ${VALA_C} libqalculatenasc/QalculateNasc.cc)
 target_link_libraries(nasc
   ${QALCULATE_LIB_PATH}
-  ${DEPS_LIBRARIES}
-  qalculatenasc)
+  ${DEPS_LIBRARIES})
 
 install (TARGETS nasc RUNTIME DESTINATION bin)
 
diff --git a/libqalculatenasc/CMakeLists.txt b/libqalculatenasc/CMakeLists.txt
deleted file mode 100644
index 776ca06..0000000
--- a/libqalculatenasc/CMakeLists.txt
+++ /dev/null
@@ -1,7 +0,0 @@
-pkg_check_modules(QALC REQUIRED libqalculate)
-link_directories(${QALC_LIBRARY_DIRS})
-include_directories(${QALC_INCLUDE_DIRS})
-
-add_library(qalculatenasc SHARED
-  QalculateNasc.cc)
-target_link_libraries(qalculatenasc ${QALC_LIBRARIES})
