diff --git a/src/MainWindow.vala b/src/MainWindow.vala
index f1ec105..42040bf 100644
--- a/src/MainWindow.vala
+++ b/src/MainWindow.vala
@@ -356,7 +356,11 @@ namespace Quilter {
             this.show_all ();
         }
 
+#if VALA_0_42
+        protected bool match_keycode (uint keyval, uint code) {
+#else
         protected bool match_keycode (int keyval, uint code) {
+#endif
             Gdk.KeymapKey [] keys;
             Gdk.Keymap keymap = Gdk.Keymap.get_for_display (Gdk.Display.get_default ());
             if (keymap.get_entries_for_keyval (keyval, out keys)) {
diff --git a/src/Widgets/Cheatsheet.vala b/src/Widgets/Cheatsheet.vala
index 33ea7d2..231f7f5 100644
--- a/src/Widgets/Cheatsheet.vala
+++ b/src/Widgets/Cheatsheet.vala
@@ -169,7 +169,11 @@ namespace Quilter.Widgets {
             }
         }
 
+#if VALA_0_42
+        protected bool match_keycode (uint keyval, uint code) {
+#else
         protected bool match_keycode (int keyval, uint code) {
+#endif
             Gdk.KeymapKey [] keys;
             Gdk.Keymap keymap = Gdk.Keymap.get_for_display (Gdk.Display.get_default ());
             if (keymap.get_entries_for_keyval (keyval, out keys)) {
